tests/common.go

```diff
@@ -799,6 +811,21 @@ func ValidatePureVolumeStatisticsDynamicUpdate(ctx *scheduler.Context, errChan .
 	})
 }
 
+// ValidatePureVolumeSnapshotAndRestore is the ginkgo spec for validating actually creating a FADA snapshot, restoring and verifying the content
+func ValidatePureVolumeSnapshotAndRestore(ctx *scheduler.Context, errChan ...*chan error) {
+	context("For validation of an snapshot and restoring", func() {
+		snapShotClassName := PureSnapShotClass + strconv.Itoa(int(time.Now().Unix()))
```

Comment:
Why you need to call `strconv.Itoa(int(time.Now().Unix())` multiple time? Your snapShotClassName, PVC name and dirtyData string will have different timestamp. They are in a single test. The timestamp should be the same.